docs/wiki/01 - STUDY PARAMETERS SETUP/01 - Study Variables/01 - Identifiers.md

```diff
@@ -0,0 +1,79 @@
+# Identifiers
+
+## Overview
+By default LORIS provides 2 different identifiers for each candidate: 
+
+- **CandID**, also known as the **DCCID**, is a unique randomized 6-digit numeric ID (e.g. '436792') assigned automatically by LORIS upon participant registration. The CandID is not configurable.
+- **PSCID** (Project Study Center ID) is a unique configurable ID. It can be be setup to be either manually entered when registering a participant or automatically generated and usually contains the site abbreviation followed by sequential or randomized characters (e.g. 'MTL0006'), but its exact format is customizable.
+
+The format and the generation of the **PSCID** must be configured by an admin at the setup stage.
+
+## Configuration
+
+### Configuration from the front end
+_not yet available_
+
+### Configuration from SQL
+_not yet available_
+
+### Configuration from the config.xml file
+
+Both the format and the generation of PSCIDs can be configured by the project in the `config.xml` file of the `%LORIS_ROOT%/project/` directory.
+
+PSCIDs can be created for new subjects in one of 3 ways: *sequentially generated*, *manually entered*, or *randomly generated*.
+
+1. ***sequential*** generates PSCIDs sequentially for each new participant registered. **(default)**
+
+ ```xml
+ <PSCID>
+     <generation>sequential</generation> 
+     <structure>
+         <seq type="siteAbbrev"/>
+         <seq type="numeric" minLength="2" maxLength="4" min="10" max="9999"/>
```

Comment:
I think it would help to give an example. 
the min length is 2, maxLength 4, and the min value is 10
 does this mean MTL0001, MTL0002 is not possible?

Reply:
Im not sure of the behaviour so maybe I'll let @johnsaigle or @PapillonMcGill comment here.

for example, given that setting is the first ID generated `MTL10` or `MTL0010`?
is it wise to allow a min/max ? shouldnt we just define a fixed length ?
if `MTL10` is generated, does it also generate `MTL0010` when it gets to length 4 ? or is it smart enough to account for trailing 0s ??

Reply:
PSCID will got from MTL10 to MTL9999 in that case.

Reply:
Having a minlength and maxLength would create problem for generation of the next PSCID value. Next value is created using max(PSCID).
PSCID MTL100 would be less (in value) than MTL99. I believe that the min length should be equal to the length of max (4 in that case) to prevent that kind of problem. Unless the length of PSCID is taken into account in the query, which would make it not simple.

Reply:
After verification, if the sequence is numerical, there is no problem (the prefix is removed before finding the max value). But if the sequence is alphanumerical (or alpha) then "Z" would be selected as max over "AA". Thus the code will try to create "AA" even if already existing.

@ridz1208 @driusan How do we modify? Do we require minLength and maxLength to be the same value (fix length) or modify IDGeneration code to fix this problem?

Reply:
For now I will stop modifying this document until we reach a solution. This is a bug not having anything to do with this doc.

My personal opinion, make it a fixed length for numerics, min/max lenghts dont make sense since trailing 00s dont change the value of numbers and ultimately its confusing to have both MTL99 and MTL0099 in one database

Reply:
I would go for fixed length even for alpha or alphanumeric, not just numerical value, to prevent ordering problem.

Reply:
i can live with that

Reply:
Good news! It already has to be a fixed length. See my review below.

Reply:
>  since trailing 00s dont change the value of numbers 

???

Reply:
@driusan A note from McGill HR re: salaries